From 26f96a178f8a0afded00bdd7238728c0b6e42a6b Mon Sep 17 00:00:00 2001
From: Richard Purdie <richard.purdie@linuxfoundation.org>
Date: Thu, 9 May 2024 18:44:41 +0000
Subject: [PATCH] meson.build: fix build race when building GTK vapi files

There's a build race when building the GTK vapi files:

FAILED: libportal/libportal-gtk4.vapi
error: Package `libportal' not found in specified Vala API directories or GObject-Introspection GIR directories

This can be verified by adding "sleep 10;" to the command for the
libportal/libportal.vapi target in the generated build.ninja file.

The GTK vapi files need to have access to the generic libportal.vapi file,
but there is no explicit dependency.  Switch the dependency name 'libportal'
to the dependency object libportal_vapi so that Meson generates the
dependency correctly.

Upstream-Status: Backport
Signed-off-by: Ross Burton <ross.burton@arm.com>
---
 libportal/meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libportal/meson.build b/libportal/meson.build
index fff7603..4e67f40 100644
--- a/libportal/meson.build
+++ b/libportal/meson.build
@@ -168,7 +168,7 @@ if gtk3_dep.found()
     if vapi
       libportal_gtk3_vapi = gnome.generate_vapi('libportal-gtk3',
         sources: libportal_gtk3_gir[0],
-        packages: ['gio-2.0', 'gtk+-3.0', 'libportal'],
+        packages: ['gio-2.0', 'gtk+-3.0', libportal_vapi],
         gir_dirs: [meson.current_build_dir()],
         vapi_dirs: [meson.current_build_dir()],
         install: true,
@@ -227,7 +227,7 @@ if gtk4_dep.found()
     if vapi
       libportal_gtk4_vapi = gnome.generate_vapi('libportal-gtk4',
         sources: libportal_gtk4_gir[0],
-        packages: ['gio-2.0', 'gtk4', 'libportal'],
+        packages: ['gio-2.0', 'gtk4', libportal_vapi],
         gir_dirs: [meson.current_build_dir()],
         vapi_dirs: [meson.current_build_dir()],
         install: true,
-- 
2.34.1

